Create Youtube Video Viewing Homepage.

cod3smith opened the pull request:
#### What does this PR do?
- This PR contains the code for creating a simple Youtube video homepage. It uses **CSS Flexbox** and **CSS Grid** to accomplish this. It also demonstrates the use of semantic HTML tags from **HTML5**

#### How would you run this project?
- Clone the project.
- Run the `index.html` file on your browser.

cod3smith committed:
Create & Style navbar.

cod3smith committed:
Create Folder Structure.

btarekegn committed:
added related video section and current video description

cod3smith committed:
Add comments section.

btarekegn committed:
added readme file

Dandush03 commented:
# :red_circle: Status Require: changes :muscle: 
### Hey :wave: 
## Good attempts on this one :muscle: :+1: 

## Here are the issues to address to get the project meet the requirement:wink:

# Description

## Project Specification
Don't forget that this project consists of building an HTML document that matches the appearance of this [YouTube video player page](http://archive.vn/Bss88).

Check it [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/48945837/)

-------------------------------------------

### Page structure

-------------------------------------------
#### Navigation bar

##### Left Part
- [ ] Logo is too small
- [ ] Hamburger button is too big

##### Center
- [ ] Input should be wider
- [ ] Search button missing padding
- [ ] Search Button missing same border as input
- [ ] Search Button need a more clear background

##### Right
- [ ] Buttons should be aligned to the right
- [ ] There should be only an upload button and a blue sign in one

-------------------------------------------

#### Require:
![Screenshot from 2020-05-07 08-14-35](https://user-images.githubusercontent.com/51087936/81295265-2a48af00-903e-11ea-902c-9cbb1d4de502.png)

#### Yours:
![Screenshot from 2020-05-07 08-14-00](https://user-images.githubusercontent.com/51087936/81295275-2d439f80-903e-11ea-8b9b-f7543daaa168.png)

-------------------------------------------

#### Video

##### Title (Aligned to the Left)

- [ ] It should contain a hero profile picture
- [ ] It should contain a hero name
- [ ] It should contain a subscribe button
- [ ] It should contain the number of views (aligned to the right)

##### Title Buttons

- [ ] Missing "Add to", "Share", "More", "Like" and "Dislike" buttons

##### Description

- [ ] It should contain the published date

-------------------------------------------

#### Require:
![Screenshot from 2020-05-07 08-20-45](https://user-images.githubusercontent.com/51087936/81295297-33d21700-903e-11ea-97f2-f896f9a012d3.png)

#### Yours:
![Screenshot from 2020-05-07 08-21-00](https://user-images.githubusercontent.com/51087936/81295304-36347100-903e-11ea-8acb-899691ace342.png)

-------------------------------------------

### Sidebar Recommended videos

- [ ] Missing Titles ("Up Next" and "Autoplay")
- [ ] Description is too big
- [ ] It Should show the numbers of views

-------------------------------------------

#### Require:
![Screenshot from 2020-05-07 08-28-38](https://user-images.githubusercontent.com/51087936/81295346-451b2380-903e-11ea-9fae-998692f09766.png)

#### Yours:
![Screenshot from 2020-05-07 08-28-54](https://user-images.githubusercontent.com/51087936/81295354-46e4e700-903e-11ea-8756-b7eade2ee4c9.png)

-------------------------------------------

### Additional Microverse Requirements (Pathwright)

#### General

- [ ] Check HTML markup with this [validator](https://validator.w3.org/).([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/validation_requirement.png) from Pathwright)

-------------------------------------------

#### Pic:
![Screenshot from 2020-05-07 08-06-52](https://user-images.githubusercontent.com/51087936/81295365-4ba99b00-903e-11ea-8bb4-fa280a24862d.png)



## Kindly address all these changes and submit a new code review


### Happy coding:computer: 

#### Daniel Laloush

cod3smith committed:
Fix Navbar.

cod3smith committed:
Merge branch 'develop' of https://github.com/KelynPNjeri/Youtube-Homepage into develop

cod3smith committed:
Add vector graphics.

btarekegn committed:
Fix comments on related videos

Added three elements Up next link, autoplay and info icons.

it's addditonal features

cod3smith committed:
Add a script tag
It will allow us to link the HTML to the JS
Linking JS
Allow access to JS.

cod3smith committed:
Merge branch 'develop' of https://github.com/KelynPNjeri/Youtube-Homepage into develop

cod3smith committed:
Fix the video section.
This commit would add more functionalities i.e sharing, adding to playlist and other features to the pre-existing design.
This improves on the overall design of the page.
Yes, it restructures the outlook of the page.

btarekegn committed:
Fix positioning of elements
Rearranged the position of the current video description.
The change is introduced to match the design layout of the youtube home page

cod3smith committed:
Fix the malformed profile image avatars.
It improves on the general outlook of the webpage.
It changes the widths and heights of the images to make them look alright.
Improves ui/ux of the page.

cod3smith committed:
Fixing the related videos section.
This creates a better appeal of the page.
It addresses the issue of a long video description and missing video views and duration since upload.
This includes a way a user of the platform can view a short description of the vido plus who the youtuber was.

cod3smith committed:
Resolve merge conflict.

btarekegn committed:
Fix video action alignment

bug introduced during refactoring other elements

cod3smith committed:
fix the issues brough up by TSE.
The commit fixes the icon alignment issues, duplicate about video description and lack of divider on first video..
This commit improves the general UI/UX of the entire page.

cod3smith committed:
Add a well structured README to thr project.
This provides clear guidelines on what the project is about and how to set it up if interested in viewing it or collaborating.
It creates a document where people can get more info about what the project is about.

btarekegn committed:
added berhanu as contributor

berhanu-tarekegn commented:
## Updated PR
- We updated the project based on the code review comments given previously.
- Validated the **CSS** and the **HTML** using [W3 Validator](https://validator.w3.org/)
- We managed to add the readme section based on **microverse** template 

@Dandush03 a kind reminder here, could you please take a look at.

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,273 @@
+.center {
+    width: 100%;
+    display: flex;
+    align-items: center;
```
CSS bad practice: no repeating of the same rule
```
.left,
.right,
.share {
    display: flex;
    align-items: center;
}
.center,
.current_video,
.comments {
    width: 100%;
}
.share {
    margin: auto
}
```

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,273 @@
+.description > div > p {
+    padding-bottom: 20px;
+}
+.comments {
```
CSS bad practice: no repeating of the same rule. See first inline comment.

rammazzoti2000 commented on assets/css/style.css:
```diff
@@ -0,0 +1,273 @@
+    background: white;
+    margin-bottom: 20px;
+}
+.share {
```
CSS bad practice: no repeating of the same rule, see the first inline comment

rammazzoti2000 commented on index.html:
```diff
@@ -0,0 +1,230 @@
+            <div class="center">
+              
+                <input id="search-input" type="text" placeholder="Search something...">
+                <img src="./assets/img/search.png" alt="" id="search-btn">
```
Should be a `<button>` tag that represents the search button ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/embedding_images_and_video/search_button_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/48945837/))

rammazzoti2000 reviewed (changes requested):
### Hello @KelynPNjeri and @berhanu-tarekegn :wave:  great job this far :100:
Great job and implementing the changes required by the previous  TSE
There are still though some issues that need to be fixed before approving this project :pray: 
### :stop_sign: Require Changes :construction: :construction_worker: 
- [x] Make sure that basic CSS bad practices are not used ([article for support ](https://speckyboy.com/good-bad-css-practices)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) from Pathwright), in this case no repeating of the same rule, the following has been refactored to avoid repeating the same rule many times :point_down:, please see the inline comment
```
.left,
.right,
.share {
    display: flex;
    align-items: center;
}
.center,
.current_video,
.comments {
    width: 100%;
}
.share {
    margin: auto
}
```
- [ ] The button to the right of the input must be a `<button>` tag that represents the search button ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/embedding_images_and_video/search_button_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/48945837/)), yours is an image
- [ ] The NavBar It has at least three embedded images on the rightmost part of the navbar that represent icons and/or an avatar ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/embedding_images_and_video/icons_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/48945837/))
#### Clone: 
![image](https://user-images.githubusercontent.com/3244654/81733523-73f82600-9492-11ea-96aa-0d9978bda049.png)
#### Original:
![image](https://user-images.githubusercontent.com/3244654/81733601-91c58b00-9492-11ea-9be6-ee5434e637f2.png)
- [ ] Make the number of subscribers is aligned on the oh the subscribe button and the like/dislike buttons should be better aligned they seem to be a bit off
![image](https://user-images.githubusercontent.com/3244654/81733902-ff71b700-9492-11ea-9578-bf2f0c5497ac.png)
- [x] Also make sure to properly align you like/dislike and reply button in you comments section, and the picture must be render correctly, yours is stretched out.
#### Clone:
![image](https://user-images.githubusercontent.com/3244654/81734529-ed444880-9493-11ea-9765-7639467c06a3.png)
#### Original: 
![image](https://user-images.githubusercontent.com/3244654/81734581-ff25eb80-9493-11ea-93b0-76d9dd2aef64.png)
### <em>Kindly fix the REQUIRED issues and send another code review request.

Cheers and Happy coding! :clap: 

##### Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of alexandru if you have any additional question)</em>

### :stop_sign:  Status: Changes required.

rammazzoti2000 reviewed (changes requested):
### :stop_sign:  Status: Required Changes

cod3smith committed:
Add two more icons to the right div.
This feature adds the notification and view your apps features.

cod3smith committed:
Fix the navigation alignment issue.
This fix ties together the UI/UX aspect of the entire navbar.
It removes the align-start attribute for the logo, reduces the width of the search bar and adds two icons to the right div,
It improves the overall look and feel of the page.

cod3smith committed:
Fix alignment issue between the up next text and autoplay
This is a bug fix that ensures both the up next and autoplay are on the same line. It makes the aside content more visually appealing.
It adds a margin top to the a tag in the upnext div.
It pushes the up next text down to be aligned with the autoplay.

btarekegn committed:
refactored total comment section and subscribe counters

cod3smith committed:
Merge pull request #3 from KelynPNjeri/fix-navigation-sidebar-issues

Addressing the Issues on the Navigation bar and Sidebar.

btarekegn committed:
Merge branch 'develop' of https://github.com/KelynPNjeri/Youtube-Homepage into bug-fix-comment-section

cod3smith committed:
Merge pull request #4 from KelynPNjeri/bug-fix-comment-section

refactored total comment section and subscribe counters

KelynPNjeri commented:
## Updated PR
- [x] We updated the project based on the code review comments given previously.
- [x] We fixed the alignment issues that were present on the navbar and on the sidebar. 
- [x] We also added more icons to the right div on the navbar.
- [x] We aligned the like and dislike buttons together with text accompanying it.

santiago-rodrig commented:
# Pull Request under review by @santiago-rodrig (TSE)

santiago-dev96 commented on index.html:
```diff
@@ -0,0 +1,236 @@
+                        <div></div>
+                        
+                        <span class="subscribers"><button id="subscribe-btn">Subscribe</button>
+                            <div class="number_of_subscribers">799</div>
```
This caused a validation error

santiago-dev96 reviewed (approved):
# Project approved :100:
Good job implementing the previous TSE required changes, you meet with all the requirements for approval, you can go on with the next projects now.

Have a nice day :wave:!

btarekegn committed:
fixed validation errors
